src/tox/config/__init__.py

```diff
@@ -1864,6 +1878,12 @@ def _replace_substitution(self, match):
             val = self._substitute_from_other_section(sub_key)
         if callable(val):
             val = val()
+        if isinstance(val, PathBase):
+            val = str(val)
+            # XXX handle ' and " in paths
```

Comment:
This has partially been done.
I was thinking of switching quoting char to avoid clashes with `'`.  The quoting could even be done with an invalid filename char (e.g. non-printables), as long as it makes it through the regex engine.